### PR TITLE
Signin: improvements

### PR DESCRIPTION
**Pull request by breville**
Various improvements to signin UI.

before:

screenshot 2016-08-15 13 33 17

after:

screenshot 2016-08-14 21 31 09

Some notes on the `devise/shared/_links.haml` partial.  It's included in five views, but only two of them appear to be used on our site.
- `devise/registrations/new.html.haml`
  - sign up
- `devise/sessions/new.html.haml`
  - sign in
- `devise/confirmations/new.html.haml`
  - there's a page, but GA says it gets no traffic
- `devise/passwords/edit.html.haml`
  - the /users/password/edit route works but redirects to signed-in homepage
- `devise/unlocks/new.html.haml`
  - no route

### CODE REVIEW

File: dashboard/app/views/devise/shared/_links.haml [hunk 1]
```diff
@@ -1,38 +1,28 @@
-- if controller_name != 'sessions'
+-# Unless we're already on the Sign in page, show a Sign in link
+- unless controller_name == 'sessions'
   = t('auth.already_signedup')
   = link_to t('nav.user.signin'), new_session_path(resource_name)
   %br/
   %br/
-- if devise_mapping.registerable? && controller_name != 'registrations'
-  = t('auth.notyet_signedup')
-  = link_to t('nav.user.signup'), new_user_registration_path
```
Comment (breville) [inline dashboard/app/views/devise/shared/_links.haml:8]: Now the only time we show sign-up link is done inline in the sign-in page.

File: dashboard/app/views/devise/shared/_links.haml [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
-  %br/
-  %br/
--# If we are in sign-in (but not sign-up) and can offer to recover forgotten password.
-- if controller_name == 'sessions' && devise_mapping.recoverable? && controller_name != 'passwords'
-  = link_to t('auth.forgot_password'), new_password_path(resource_name)
```
Comment (breville) [inline dashboard/app/views/devise/shared/_links.haml:13]: Now the only time we show forgot-password link is done inline in the sign-in page.

File: dashboard/app/views/devise/shared/_links.haml [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
-  %br/
-  %br/
--#
-  - if devise_mapping.confirmable? && controller_name != 'confirmations'
-  = link_to t('auth.need_confirmation'), new_confirmation_path(resource_name)
```
Comment (breville) [inline dashboard/app/views/devise/shared/_links.haml:18]: Removing commented-out code.

File: dashboard/app/views/devise/sessions/new.html.haml [hunk 4]
```diff
@@ -16,36 +16,57 @@
...
-          = t('signin_form.remember')
-      %p
-        &nbsp;
-      = f.submit t('signin_form.submit'), class: "btn btn-primary btn-large"
+        .singleitemblock
+          %label.rememberme
+            = f.check_box :remember_me
+            = t('signin_form.remember')
+
+      -# Not yet signed up?  Sign up
```
Comment (tanyaparker) [inline dashboard/app/views/devise/sessions/new.html.haml:47]: nit: extra space

File: dashboard/app/views/devise/sessions/new.html.haml [hunk 5]
```diff
@@ -55,13 +76,11 @@
 %br/
 
 :javascript
-  $( "#user_login" ).placeholder();
-
+  $("#user_login").placeholder();
 
-  $( ".edit_user" ).on("submit", function(e){
-  });
+  $(".edit_user").on("submit", function(e) {});
```
Comment (tanyaparker) [inline dashboard/app/views/devise/sessions/new.html.haml]: nit: space between `function (e)`

File: dashboard/app/views/devise/sessions/new.html.haml [hunk 6]
```diff
@@ -55,13 +76,11 @@
...
 
-  $( "#new_user" ).on("submit", function(e){
+  $("#new_user").on("submit", function (e) {
     window.dashboard.hashEmail({ email_selector: '#user_login',
       hashed_email_selector: '#user_hashed_email'});
```
Comment (tanyaparker) [inline dashboard/app/views/devise/sessions/new.html.haml]: nit: `);` on new line I think?

File: dashboard/app/views/devise/sessions/new.html.haml [hunk 7]
```diff
@@ -55,13 +76,11 @@
...
-
+  $("#user_login").placeholder();
 
-  $( ".edit_user" ).on("submit", function(e){
-  });
+  $(".edit_user").on("submit", function(e) {});
 
-  $( "#new_user" ).on("submit", function(e){
+  $("#new_user").on("submit", function (e) {
     window.dashboard.hashEmail({ email_selector: '#user_login',
```
Comment (tanyaparker) [inline dashboard/app/views/devise/sessions/new.html.haml]: nit: email_selector on new line?

**Comment by tanyaparker** [comment]
LGTM

**Comment by tanyaparker** [comment]
Also, I like to add a before screenshot too. I bet we'll eventually touch this again, and we always want to look back at what we fixed it from.

**Pull request merged**